wallet-new/src/Cardano/Wallet/Kernel/DB/TxMeta/Types.hs

```diff
@@ -204,6 +230,27 @@ data SortCriteria =
     | SortByAmount
     -- ^ Sort the 'TxMeta' by the amount of money they hold.
 
+data FilterOperation a =
+    NoFilterOp
+    -- ^ No filter operation provided
+    | FilterByIndex a
+    -- ^ Filter by index (e.g. equal to)
+    | FilterByPredicate FilterOrdering a
+    -- ^ Filter by predicate (e.g. lesser than, greater than, etc.)
+    | FilterByRange a a
+    -- ^ Filter by range, in the form [from,to]
+    | FilterIn [a]
+    deriving (Show, Eq)
```

Comment:
Why did you duplicate these rather than use the existing types?

Reply:
@edsko so that we don`t have V1 and Api types in the Kernel.